eng/docker/Mono/Dockerfile

```diff
@@ -40,6 +40,12 @@ RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E03280
     apt-get install -y mono-devel && \
     apt-get clean
 
+
+# Update previously installed mono-devel.
+# Cache bust ensures we'll always run the commands following regardless of docker cache status
+ARG CACHE_BUST=0
+RUN apt-get update && apt-get upgrade -y
```

Comment:
Me and @agocke spoke about this yesterday. We figured that we should keep everything up to date as otherwise we could get into weird states where an 'incremental' build fails differently to a new build due to package versioning diffs.

My thinking was to update all, then restrict if we see it taking too long in the future.

Reply:
Running this in the middle will disable caching for all subsequent statements. Is that what you intended?

Reply:
Yes, we edit the user permissions to make them match the outer user. That causes the upgrade/update to have issues, so it's simpler to do it before hand. The user stuff takes a trivial amount of time to run, so we aren't hurt by not caching it. (Also, we have no guarantees that the outer user will stay the same, so we'd potentially *not* want to cache it anyway, although the chance of it changing and the cache remaining seem slim)

---
In reply to: [251963406](https://github.com/dotnet/roslyn/pull/32885#discussion_r251963406) [](ancestors = 251963406)